tree/dataframe/inc/ROOT/RDFRange.hxx

```diff
@@ -0,0 +1,126 @@
+// Author: Enrico Guiraud, Danilo Piparo CERN  09/2018
+
+/*************************************************************************
+ * Copyright (C) 1995-2016, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT_RDFRANGE
+#define ROOT_RDFRANGE
+
+#include "ROOT/RRangeBase.hxx"
+#include "RtypesCore.h"
+
+#include <memory>
+
+namespace ROOT {
+
+// fwd decl
+namespace Internal {
+namespace RDF {
+namespace GraphDrawing {
+std::shared_ptr<GraphNode> CreateRangeNode(const ROOT::Detail::RDF::RRangeBase *rangePtr);
+} // ns GraphDrawing
+} // ns RDF
+} // ns Internal
+
+namespace Detail {
+namespace RDF {
+namespace RDFGraphDrawing = ROOT::Internal::RDF::GraphDrawing;
+
+template <typename PrevData>
+class RRange final : public RRangeBase {
```

Comment:
Another quite generic name with potential for later conflitcs. Same difference with header name as `RAction`.

Reply:
all our classes except a select few are in either `ROOT::Internal::RDF` or `ROOT::Detail::RDF`